metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py

```diff
@@ -0,0 +1,284 @@
+            FORMAT_TYPE,
+            COLUMN_NAME,
+            ORDINAL_POSITION,
+            IS_NULLABLE,
+            DATA_TYPE,
+            COLUMN_SIZE,
+            CREATED
+        FROM
+            (SELECT
+                VIEW_ID AS RESOURCE_ID,
+                VIEW_NAME AS TABLE_NAME,
+                PATH,
+                CASE
+                    WHEN LENGTH(SCHEMA_ID) = 0 THEN SPACE_ID
+                    ELSE SCHEMA_ID
+                END AS LOCATION_ID,
+                OWNER_ID,
+                SQL_DEFINITION AS VIEW_DEFINITION,
+                '' AS FORMAT_TYPE,
+                CREATED,
+                TYPE
+            FROM
+                SYS.PROJECT.VIEWS
+            UNION ALL
+            SELECT
+                TABLE_ID AS RESOURCE_ID,
+                TABLE_NAME,
+                PATH,
+                CASE
+                    WHEN LENGTH(SCHEMA_ID) = 0 THEN SOURCE_ID
+                    ELSE SCHEMA_ID
+                END AS LOCATION_ID,
+                OWNER_ID,
+                NULL AS VIEW_DEFINITION,
+                FORMAT_TYPE,
+                CREATED,
+                TYPE
+            FROM
+                SYS.PROJECT."TABLES"
+            ) V
+        LEFT JOIN
+            (SELECT
+                USER_ID AS ID,
+                USER_NAME AS "OWNER",
+                'USER' AS OWNER_TYPE
+            FROM
+                SYS.ORGANIZATION.USERS
+            UNION ALL
+            SELECT
+                ROLE_ID AS ID,
+                ROLE_NAME AS "OWNER",
+                'GROUP' AS OWNER_TYPE
+            FROM
+                SYS.ORGANIZATION.ROLES
+            ) U
+        ON
+            V.OWNER_ID = U.ID
+        LEFT JOIN
+        (SELECT
+            TABLE_SCHEMA,
+            TABLE_NAME,
+            COLUMN_NAME,
+            ORDINAL_POSITION,
+            IS_NULLABLE,
+            DATA_TYPE,
+            COLUMN_SIZE
+        FROM
+            INFORMATION_SCHEMA.COLUMNS
+        WHERE
+            LOCATE('{container_name}', LOWER(TABLE_SCHEMA)) = 1
+        ) C
+        ON
+            CONCAT(REPLACE(REPLACE(REPLACE(V.PATH, ', ', '.'), '[', ''), ']', '')) =
+            CONCAT(C.TABLE_SCHEMA, '.', C.TABLE_NAME)
+        WHERE
+            V.TYPE NOT IN ('SYSTEM_TABLE')
+            AND LOCATE('{container_name}', LOWER(PATH)) = 2
+        )
+        WHERE 1=1
+            {schema_pattern}
+            {deny_schema_pattern}
+        ORDER BY
+            TABLE_SCHEMA ASC,
+            TABLE_NAME ASC
+            """
+
+    QUERY_ALL_JOBS = """
+    SELECT
+        *
+    FROM
+        SYS.JOBS_RECENT
+    WHERE
+        STATUS = 'COMPLETED'
+        AND LENGTH(queried_datasets)>0
+        AND user_name != '$dremio$'
+        AND query_type not like '%INTERNAL%'
+    """
+
+    QUERY_ALL_JOBS_CLOUD = """
+        SELECT
+            *
+        FROM
+            SYS.PROJECT.HISTORY.JOBS
+        WHERE
+            STATUS = 'COMPLETED'
+            AND LENGTH(queried_datasets)>0
```

Comment:
Length is not valid on an array field, use
AND ARRAY_SIZE(queried_datasets)>0
instead

Reply:
https://github.com/datahub-project/datahub/pull/11801